OPHD/States/CrimeExecution.cpp

```diff
@@ -44,3 +47,29 @@ void CrimeExecution::stealFood(FoodProduction& structure)
 			NotificationArea::NotificationType::Warning);
 	}
 }
+
+
+void CrimeExecution::stealRefinedResources(Structure& structure)
+{
+	if (structure.storage().isEmpty())
+	{
+		return;
+	}
+
+	auto resourceIndicesWithStock = structure.storage().getIndicesWithStock();
+
+	// TODO: Rob randomly from stock, for now it just robs from first available indice with resources
+	auto indexToStealFrom = resourceIndicesWithStock[0];
+
+	int amountStolen = std::clamp(100, 0, structure.storage().resources[indexToStealFrom]);
```

Comment:
You may want to check the parameter order of [`std::clamp`](https://en.cppreference.com/w/cpp/algorithm/clamp). The value comes first. Using an alternate parameter order may appear to work in some cases, but can be problematic and throw exceptions when `low >= high`. That can potentially happen on rare occasions when variable values are passed for either or both of the later two parameters.

Reply:
I think this is correct. The goal is to steal 100 resources, but not greater than the stored resources. 

Maybe easier to read if I set amountStolen to 100, then pass it into clamp? This is a little arbitrary, since I want to use the random number generator to make the amount stolen variable in the future.

Reply:
Logically that should be equivalent to stealing all available resources, but clamping the value at 100. That would be safer, considering how `std::clamp` is implemented.

> Reference to lo if v is less than lo, reference to hi if hi is less than v, otherwise reference to v.

> The behavior is undefined if the value of lo is greater than hi.

The constants should be `lo` and `hi`. It's a safer way to use the function.

See bug #588, and fixes #589, and #590.

In particular, take a look at what the fix was:
https://github.com/OutpostUniverse/OPHD/pull/589/commits/27ddeb366cf47d4f67d208f1f8efa4aed0a9b05b

Reply:
I guess my assumption was the amount in storage was always 0 or positive. If the amount in storage is negative, it would maybe be good to throw an exception? 

I'm pulling std::clamp since it is fairly easy to implement without the standard library call, which will remove the contention.

Reply:
I suppose my other concern is the code will serve as an example to be copied later, where assumptions will be different, and potentially more dangerous.

As for throwing an exception, this would be the wrong place for that exception. It would be confusing to get the exception here, rather than the code location that sets the negative value. From that perspective, your code is absolutely correct to assume the value should be non-negative. If we throw an error, it should be done elsewhere. Throwing the exception here would lead to a confusing debugging experience.